contracts/script/releases/v2.4.1-access-control-migration/2-ExecuteUpgrade.s.sol

```diff
@@ -0,0 +1,322 @@
+        assertTrue(Env.proxy.paymentVault().pricePerSymbol() > 0, "PaymentVault: pricePerSymbol not preserved");
+        assertTrue(Env.proxy.paymentVault().priceUpdateCooldown() > 0, "PaymentVault: priceUpdateCooldown not preserved");
+        assertTrue(Env.proxy.paymentVault().globalSymbolsPerPeriod() > 0, "PaymentVault: globalSymbolsPerPeriod not preserved");
+        assertTrue(Env.proxy.paymentVault().reservationPeriodInterval() > 0, "PaymentVault: reservationPeriodInterval not preserved");
+        assertTrue(Env.proxy.paymentVault().globalRatePeriodInterval() > 0, "PaymentVault: globalRatePeriodInterval not preserved");
+    }
+
+    /// @notice Verify ThresholdRegistry state was preserved during upgrade
+    function _testThresholdRegistryStatePreservation() internal view {
+        assertTrue(Env.proxy.thresholdRegistry().quorumAdversaryThresholdPercentages().length > 0, 
+            "ThresholdRegistry: quorumAdversaryThresholdPercentages not preserved");
+        assertTrue(Env.proxy.thresholdRegistry().quorumConfirmationThresholdPercentages().length > 0, 
+            "ThresholdRegistry: quorumConfirmationThresholdPercentages not preserved");
+        assertTrue(Env.proxy.thresholdRegistry().quorumNumbersRequired().length > 0, 
+            "ThresholdRegistry: quorumNumbersRequired not preserved");
+    }
+
+    /// @notice Verify critical references were preserved during upgrade
+    function _testCriticalReferencesPreserved() internal view {
+        assertTrue(Env.proxy.registryCoordinator().ejector() != address(0), "RegistryCoordinator: ejector not preserved");
+        assertTrue(Env.proxy.serviceManager().rewardsInitiator() != address(0), "ServiceManager: rewardsInitiator not preserved");
+    }
+
+    /// @notice Verify PauserRegistry configuration is correct
+    function _testPauserRegistryConfiguration() internal view {
+        // assertEq(address(Env.proxy.registryCoordinator().pauserRegistry()), address(Env.impl.pauserRegistry()), 
+        //     "RegistryCoordinator: incorrect pauserRegistry");
+        // assertEq(address(Env.proxy.serviceManager().pauserRegistry()), address(Env.impl.pauserRegistry()), 
+        //     "ServiceManager: incorrect pauserRegistry");
+    }
+
+    /// @notice Verify Directory has access control configured
+    function _testDirectoryAccessControl() internal view {
+        address accessControlAddr = Env.proxy.directory().getAddress(keccak256("ACCESS_CONTROL"));
+        assertTrue(accessControlAddr != address(0), "Directory: accessControl not set");
+        assertEq(accessControlAddr, address(Env.impl.accessControl()), 
+            "Directory: incorrect accessControl address");
+    }
+
+    /// @notice Verify cross-contract references are still correct
+    function _testCrossContractReferences() internal view {
+        assertEq(address(Env.proxy.serviceManager().avsDirectory()), address(Env.proxy.avsDirectory()), 
```

Comment:
shouldn't we also be checking the service manager's reference to:
- rewards coordinator
- pauser registry
- stake registry
- registry coordinator

<img width="849" height="672" alt="Image" src="https://github.com/user-attachments/assets/07c154a1-8910-4e14-90e7-071bd0bb0d1c" />

also should we validate that what the service manager points matches the associated named keys that have been populated in the contract directory?